[Payments] Add WooPayments IPP support for Puerto Rico

iamgabrielma opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #14646
<!-- Id number of the GitHub issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

## Description
As with the [Android counterpart](https://github.com/woocommerce/woocommerce-android/pull/13200) this PR adds IPP support for Puerto Rico by not blocking nor preventing to process a card payment in the mobile client when the store address is set to `PR`.

Since the fix to support this state needs WooPayments 9.0.0, and TTP doesn't seem to be supported (p1741684540420039-slack-C025A8VV728) I've made a separate configuration rather than reusing `US`.

## Testing information
You can use the test store here:  37357-pb , or one of your stores, as long as runs WooComment Payments 9.0 and its address is set to Puerto Rico.
- Attempt to process an IPP payment
- Observe on available payment methods that the `Card Reader` row is rendered, but `Tap to Pay` row is not rendered
- Tap on `Card Payment` and process an IPP payment, observe that works as expected.

https://github.com/user-attachments/assets/4429e8ee-ad74-4bfa-9fc3-68ef22d55c12

In the same nature, there is no specific `/PR/` URL to acquire a card reader, upon navigating to `Payments` > `Order Card Reader` it will fallback to the M2 reader:

<img width=350 src="https://github.com/user-attachments/assets/7e877e0e-39dd-408e-bc06-76868d1bd43e"/>

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

iamgabrielma committed:
add IPP support for Puerto Rico using US config

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
update release notes

staskus commented:
Unfortunately, it doesn't work for me as well. I tried to:

- Change existing store address to Puerto Rico
- Create a new store with Puerto Rico address and onboard to WooPayments with Puerto Rico address

Both result in connection failures.

```
💳 Stripe Error Code: SCPError(rawValue: 9020)
💳 Card Reader Service Error: connection(underlyingError: Hardware.UnderlyingError.processorAPIError)
CoreData: debug: PostSaveMaintenance: incremental_vacuum with freelist_count - 9 and pages_to_free 1
🔵 Tracked card_reader_discovery_failed, properties: [is_wpcom_store: false, was_ecommerce_trial: false, plan: jetpack_free, site_url: https://obvious-toad-bobolink.jurassic.ninja, site_id: 241124518, country: PR, store_id: e783a36f-e98e-4cf2-ac73-bced35b3c4be, error_description: Sorry, this payment couldn’t be processed., connection_type: user_initiated, plugin_slug: woocommerce-payments, blog_id: 241124518]
🔵 Tracked card_reader_connection_failed, properties: [is_wpcom_store: false, was_ecommerce_trial: false, card_reader_model: STRIPE_M2, plan: jetpack_free, site_id: 241124518, site_url: https://obvious-toad-bobolink.jurassic.ninja, country: PR, store_id: e783a36f-e98e-4cf2-ac73-bced35b3c4be, error_description: The payment can not be processed by the payment processor., connection_type: user_initiated, plugin_slug: woocommerce-payments, blog_id: 241124518]
```

There could be an additional step missing to make it work 🤔

iamgabrielma commented:
Thanks for testing, the [9020](https://github.com/stripe/stripe-terminal-ios/blob/master/PublicHeaders/SCPErrors.h) seems to indicate an API error, but it's pretty generic. I'll take a further look 🤔 
```
 /**
     The underlying request returned an API error.
     */
    SCPErrorStripeAPIError = 9020,
```

iamgabrielma commented:
Just an update @staskus : We have a fix in WooCommerce Payments to resolve this issue that I tested last week but felt out of priority due POS work. I'll circle back to this PR either on Monday or post-meetup 👍

iamgabrielma committed:
Merge branch 'trunk' into task/14646-add-ipp-support-for-puerto-rico

iamgabrielma committed:
Make PR configuration separate from US

Since PR explicitely requires WCpay 9.0+ and does not currently support TTP we cannot reuse the US config

iamgabrielma committed:
Update tests

iamgabrielma commented:
This is ready for review @staskus , I've updated the testing instructions with a test store already set to PR 👍

staskus committed:
Merge branch 'trunk' into task/14646-add-ipp-support-for-puerto-rico

staskus commented on Yosemite/Yosemite/Model/Payments/CardPresentPaymentsConfiguration.swift:
```diff
@@ -62,6 +62,22 @@ public struct CardPresentPaymentsConfiguration: Equatable {
+                supportedReaders: [.chipper, .stripeM2],
+                supportedPluginVersions: [
+                    .init(plugin: .wcPay, minimumVersion: "9.0.0"),
+                    .init(plugin: .stripe, minimumVersion: "6.2.0")
```
Do we really support Stripe or should we apply the same fix as we did with WooPayments?

```suggestion
```

On your test store, I added a Stripe extension but the card reader connection fails.

iamgabrielma replied:
> Do we really support Stripe or should we apply the same fix as we did with WooPayments?

That's a good question, I've only scoped these changes to WooCommerce Payments. I'm unsure who's currently tackling changes in the Stripe plugin and if we should handle this as well, I've asked here: p1741748917720889?thread_ts=1739205177.793219&cid=CGGCLBN58-CGGCLBN58

iamgabrielma replied:
After several attempts this last week with wcpay folks ( p1739205177793219-ref-CGGCLBN58 ) it seems that the Stripe plugin will need dedicated time to make this work, I've [opened an issue on the plugin repo](https://href.li/?https://github.com/woocommerce/woocommerce-gateway-stripe/issues/4061) and removed support for the Stripe plugin in this PR: https://github.com/woocommerce/woocommerce-ios/pull/14972/commits/b457b4f71b592cfe7f1f70770486d378d7a67564

staskus reviewed (commented):
Thank you!

PR payments work with the card reader using WooPayments. However, it doesn't work with the Stripe extension.

iamgabrielma committed:
remove Stripe support from PR configuration

The Stripe plugin doesn’t handle terminal location and we still see the 9020 error, this will be handled separately on the plugin side before we can add support for it in mobile

iamgabrielma committed:
Merge branch 'task/14646-add-ipp-support-for-puerto-rico' of https://github.com/woocommerce/woocommerce-ios into task/14646-add-ipp-support-for-puerto-rico

staskus commented on RELEASE-NOTES.txt:
```diff
@@ -3,6 +3,8 @@
 
 22.0
 -----
+- [*] Payments: Puerto Rico is now available in the list of countries that are supported by In-Person Payments, when using WooPayments [https://github.com/woocommerce/woocommerce-ios/pull/14972]
```
For some reason, we have added two very similar release notes. Maybe the result of some conflict resolution? Let's remove one of them 👍

iamgabrielma replied:
Good catch 🤦 , yes, it seems to come from a conflict between the latest 2 trunk merges, as in one of them we updated the note and didn't catch it as a conflict. Fixed: 6430c384bea4e13e751880bf6f6b6b5b3007811f

staskus reviewed (approved):
Thanks for following through! 👍 

Maybe we can even update PR name to be clear that it only applies to WooPayments.

iamgabrielma committed:
Update RELEASE-NOTES.txt

Remove duplication from merge conflict
